File: Python/TarjetaDeDebito.py
from IEstado import IEstado
from Normal import Normal

class TarjetaDeDebito:

    __nombre: str
    __numero: str
    __estado: IEstado = Normal()
    #nombre;
    #numero;
    #estado = new Normal();

    def __init__(self, nombre, numero):
        self.__nombre = nombre
        self.__numero = numero

    def retirarSaldo(self, saldoARet: float):
        self.__estado.retirarSaldo(saldoARet)

    def cambiarEstado(self, estado: IEstado):
        self.__estado = estado

File: NodeJS/TarjetaDeDebito.js
const { Normal } = require("./State/Normal");

class TarjetaDeDebito{

    #nombre;
    #numero;
    #estado = new Normal();

    constructor(nombre, numero){
        this.#nombre = nombre;
        this.#numero = numero;
    }

    retirarSaldo(saldoARet){
        this.#estado.retirarSaldo(saldoARet);
    }

    cambiarEstado(estado){
        this.#estado = estado;
    }

}

exports.TarjetaDeDebito = TarjetaDeDebito;